File: examples/scala/mega_flow/Executor.scala
package mega

import scala.sys.process._

object Executor {
  // SINK — scala.sys.process run · scala.cmdi.process_run · CWE-78
  def execute(cmd: String): String = {
    cmd.!
    cmd
  }

  def cleanTwin(): String = {
    // NEGATIVE — same sink kind with a constant argument must not report.
    "echo clean".!
    "clean"
  }
}

File: examples/php/mega_flow/executor.php
<?php
class Executor {
    // SINK — shell_exec · php.cmdi.shell_exec · CWE-78
    public static function execute(string $cmd): string {
        shell_exec($cmd);
        return $cmd;
    }

    public static function cleanTwin(): string {
        // NEGATIVE — same sink kind with a constant argument must not report.
        shell_exec('echo clean');
        return 'clean';
    }
}
